### PR TITLE
Bookmarking

### PR DESCRIPTION
**Pull request by peterbe**
This is a start on the Plus bookmarking app. 

On every document...
Screen Shot 2021-06-14 at 7 29 24 AM
Screen Shot 2021-06-14 at 7 29 30 AM
(it looks a bit weird because I don't have the other languages enable so no "Change language" link in the breadcrumb bar)

On a dedicated dashboard for all your Plus bookmarks:
Screen Shot 2021-06-14 at 7 29 49 AM

Features:

- Lazy loaded so no extra JS for people who are not subscribers (aka. paying members)
- Can cope with server errors but UI a bit rough
- Not real-time but updates on page visibility API
- When returning to previously bookmarked pages, it says when it was bookmarked when hovered over
- When clicked, does not say "Bookmarked seconds ago" until the server has confirmed successfully saved the bookmark

Not-features:

- The dashboard that lists all your bookmarks does not have pagination, filtering, search, sort order
- Might be smart to use `sessionStorage` as a cache-while-waiting-for-XHR like we do for the `whoami` in the `useUserData` hook.
- There is no link anywhere to your MDN Plus Bookmarks dashboard. You have to manually type in http://localhost.org:3000/en-US/plus/bookmarks/ for now. 
- Styling but some example SCSS to demonstrate the right and available selectors and files for where to put styling.

### CODE REVIEW

**Comment by peterbe** [comment]
@schalkneethling @escattone Let's leave this as a draft until https://github.com/mdn/kuma/pull/7903 merges. At least. 
My goal and hope are that it becomes easy for Schalk to work on this. He should be able to just use the regular `master` branch of `kuma` without many hoops. 
The truth is, even if https://github.com/mdn/kuma/pull/7903 lands in `master` you'd still need to fake around with the auth and "is-subscriber flag" because we don't have FxA or the subscription platform figured out. But we can't wait forever for that either. It would be prudent to get started on properly styling the bookmarks in advance rather than a rushed job when all the third-party stuff comes together.

**Review by escattone — COMMENTED** [review]
Looks good so far. More tomorrow.

File: client/src/plus/bookmarks/index.tsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+          <span>☆</span>
+        </button>
+      </p>
+    </div>
+  );
+}
+
+function Breadcrumbs({ parents }: { parents: Breadcrumb[] }) {
+  return (
+    <ol className="breadcrumbs">
```
[the remark this answers is not in the document]
Comment (peterbe) [reply client/src/plus/bookmarks/index.tsx:312]: > It seems like this is something that can be left for @schalkneethling in the styling phase?

Yeah, inspect the DOM in the regular breadcrumb on a page like https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video and you'll see the little "chevron" arrows are part of the CSS.

**Review by escattone — APPROVED** [review]
Let's merge this as is, and @schalkneethling can take over from here. Thanks @peterbe!

**Comment by peterbe** [comment]
I made it work with both `plus/index.tsx` *and* `plus-v1/index.tsx`. 

I've tested it a bunch and it seems to work. It also creates a skeleton SPA for `/en-us/plus/bookmarks` so that it loads nicer when you don't use the webpack dev server.

**Pull request merged**